### DISCUSSION TITLE
Support Bluetooth Keyboards

### DISCUSSION THREAD

**Discussion by lots0logs** [zbm-dev/zfsbootmenu#530] [General]
I'm not sure what this would entail or if it's even something you would consider adding. Please let me know your thoughts. Thanks.

**Comment by zdykstra** [comment]
This isn't something that I want to add to binary releases. However, we'd welcome documentation for enabling it with Dracut and mkinitcpio-based images.

**Comment by ahesford** [comment]
The biggest challenges with Bluetooth are probably 1) managing the lifetime of the Bluetooth daemon, and 2) maintaining the pairing database that would actually allow the keyboard to work. The lifetime problem is relatively easy: just fire off the daemon using, *e.g.*, a ZFSBootMenu early-setup hook or something like `mkinitcpio-rclocal`.  The bigger problem is that you'd have to make sure the Bluetooth daemon has whatever it needs to talk to the keyboard, which probably means copying things like `/var/lib/bluetooth` from your host into the image. I strongly suspect there is no way to make a generic image that works conveniently with Bluetooth.

**Reply by lots0logs** [reply]
I'll give it a try and see what I can or can't get working and keep you posted. Thanks.

**Reply by ghost** [reply]
For Arch users, I got my Bluetooth keyboard working with the `mkinitcpio-bluetooth` package in AUR ([report](https://github.com/irreleph4nt/mkinitcpio-bluetooth/issues/17#issuecomment-1987247719)). This might be hardware-dependent but I didn't have to add any additional firmware files for Intel AX200.

**Reply by ahesford** [reply]
`mkinitcpio-bluetooth` pulls a LOT of extra junk unlikely to be related to the intended task, and also pulls host-specific configs. Still, if it gets the job done for a custom ZBM image, have at it.
